cpr/session.cpp

```diff
@@ -390,6 +391,13 @@ void Session::SetConnectTimeout(const ConnectTimeout& timeout) {
     curl_easy_setopt(curl_->handle, CURLOPT_CONNECTTIMEOUT_MS, timeout.Milliseconds());
 }
 
+void Session::SetConnectionPool(const ConnectionPool& pool) {
+    auto * curl = curl_->handle;
+    if (curl) {
+        pool.SetupHandler(curl);
+    }
```

Comment:
You can expect `curl` to be valid here. If you do not trust it, use a `assert(curl)`